Chessington.GameEngine/Square.cs

```diff
@@ -49,5 +51,10 @@ public override string ToString()
         {
             return string.Format("Row {0}, Col {1}", Row, Col);
         }
+
+        public bool IsInbound()
+        {
+            return Enumerable.Range(0, 8).Contains(this.Col) && Enumerable.Range(0, 8).Contains(this.Row);
```

Comment:
You can just check if it is >= 0 and < 8. Using range is a bit over-engineering for this